src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+    private Node<T> newNode;
+    private Node<T> temporaryNode;
+
+    private static class Node<T> {
+        protected T element;
+        protected Node<T> prev;
+        protected Node<T> next;
+
+        protected Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
+    public MyLinkedList() {
+        first = new Node<>(null, null, null);
+        last = new Node<>(null, null, null);
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            temporaryNode = new Node<>(first, value, last);
```

Comment:
we can use null as parameters instead of first and last

Reply:
logic of my code must to do only with that variables!

Reply:
this if can be simplified up to 3 lines of code. or even more: try to think about checking not `== 0`, but `> 0`